packages/mds-compliance-service/repository/index.ts

```diff
@@ -0,0 +1,264 @@
+import { getManager } from 'typeorm'
+
+import { InsertReturning, RepositoryError, ReadWriteRepository } from '@mds-core/mds-repository'
+import { isDefined, NotFoundError, now } from '@mds-core/mds-utils'
+import { UUID } from '@mds-core/mds-types'
+import {
+  ComplianceSnapshotDomainModel,
+  GetComplianceSnapshotsByTimeIntervalOptions,
+  GetComplianceSnapshotOptions,
+  GetComplianceViolationPeriodsOptions,
+  ComplianceViolationPeriodEntityModel
+} from '../@types'
+import { ComplianceSnapshotEntityToDomain, ComplianceSnapshotDomainToEntityCreate } from './mappers'
+import { ComplianceSnapshotEntity } from './entities/compliance-snapshot-entity'
+import migrations from './migrations'
+
+export class SqlVals {
+  public vals: (string | number | string[])[]
+
+  private index: number
+
+  public constructor() {
+    this.vals = []
+    this.index = 1
+  }
+
+  public add(value: string | number | string[]): string | number {
+    this.vals.push(value)
+    const literal = `$${this.index}`
+    this.index += 1
+    return literal
+  }
+
+  public values(): (string | number | string[])[] {
+    return this.vals
+  }
+}
```

Comment:
There must be a native TypeORM way to do what this is being used for, no?

Reply:
Well, due to the complexity of this query, the usual query builder methods aren't usable.